packages/components/src/components/SquiggleEditorImportedBindings.tsx

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import { SquiggleEditor } from "./SquiggleEditor";
+import type { SquiggleEditorProps } from "./SquiggleEditor";
+import { runPartial, defaultBindings } from "@quri/squiggle-lang";
+import type { result, errorValue, bindings } from "@quri/squiggle-lang";
+
+function resultDefault(x: result<bindings, errorValue>): bindings {
+  switch (x.tag) {
+    case "Ok":
+      return x.value;
+    case "Error":
+      return defaultBindings;
+  }
+}
+
+function replaceBindings(
+  props: SquiggleEditorProps,
+  newBindings: bindings
+): SquiggleEditorProps {
+  return { ...props, bindings: newBindings };
+}
+
+export type SquiggleEditorImportedBindingsProps = SquiggleEditorProps & {
+  bindingsImportFile: string;
+};
+
+export const SquiggleEditorImportedBindings: React.FC<
+  SquiggleEditorImportedBindingsProps
+> = (props) => {
+  const [bindingsResult, setBindingsResult] = React.useState({
+    tag: "Ok",
+    value: defaultBindings,
+  } as result<bindings, errorValue>);
+  React.useEffect(() => {
+    async function retrieveBindings(fileName: string) {
+      let contents = await fetch(fileName).then((response) => {
+        return response.text();
+      });
+      setBindingsResult(runPartial(contents));
```

Comment:
There could be other props which would affect the execution which you don't pass to `runPartial` here, and you probably should.

Btw, would it ever be necessary to have the different `environment` or `jsImports` for imported bindings execution? I hope not, but it's something to consider.